pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -193,6 +194,29 @@ func validateSystemRequirements(mountUtil mount.Interface) (features, error) {
 	return f, nil
 }
 
+// Checks if any swap spaces are on.
+// Passes silently if /proc/swaps does not exist, e.g. on Chromebooks.
```

Comment:
I think it should only be ignore if `--fail-swap-on` is set to false.
In your case (on chromebook), couldn't you just disable the check with --fail-swap-on=false?

Reply:
/cc @derekwaynecarr @dashpole

Reply:
If the file doesn't exist, do we consider that to be "swap is on" or "swap is off"?
@therc, do chromebooks (or other OS without that file) have the ability to enable swap?

Reply:
If we are doing development workflows on a localhost, also just set the flag manually.  I would be curious if that would also work.

Reply:
It's complicated. All Chromebooks use swap, in reality, but it's not the usual beast. They use Linux' zram support to swap to a compressed ramdisk, not to actual disk. Kinda like http://www.drdobbs.com/parallel/inside-softram-95/184409937, only it's real.

Crostini uses LXD to spawn containers on ChromeOS. That's where minikube would run, and my understanding is that it turns swap off for those (see e.g. https://github.com/lxc/lxd/issues/3444). That makes sense, if we assume that zram was adopted mainly to compress Chrome's data, i.e. lots of inactive tabs.
I know about the flag, but this change would help things work out of the box. I don't know of other systems where the file would be missing, unless you compile a kernel without swapfile support.

(I don't work on ChromeOS, I am only trying to get rid of my Mac.)

Reply:
Since kubelet has no idea whether swap is on or not, I think using the flag to disable the feature explicitly still makes more sense.